packages/@aws-cdk/aws-stepfunctions-tasks/test/athena/start-query-execution.test.ts

```diff
@@ -153,4 +154,29 @@ describe('Start Query Execution', () => {
       },
     });
   });
+
+  test('omit QueryExecutionContext if no catalog or database name provided', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+
+    // WHEN
+    const task = new AthenaStartQueryExecution(stack, 'Query', {
+      queryString: 'CREATE DATABASE database',
+      clientRequestToken: 'unique-client-request-token',
+      resultConfiguration: {
+        outputLocation: {
+          bucketName: 'query-results-bucket',
+          objectKey: 'folder',
+        },
+      },
+    });
+
+
+    // THEN
+    expect(stack.resolve(task.toStateJson())).toMatchObject({
+      Parameters: {
+        QueryExecutionContext: ABSENT,
+      },
+    });
```

Comment:
hm sorry, this doesn't seem to be doing what I want. can you change to the following to see if that works better?
```suggestion
    expect(stack.resolve(task.toStateJson())).not.toMatchObject({
      Parameters: {
        QueryExecutionContext: expect.anything(),
      },
    });
```

Reply:
I just tested that - does not work. New approach is ` expect(stack.resolve(task.toStateJson())).not.toHaveProperty("Parameters.QueryExecutionContext")`
- testing that right now

Reply:
I see my error - I missed the call to the actual `anything` method and used it as property.
But I think the solution via .not.toHaveProperty looks better

Reply:
it works :)